backend/pkg/commons/utils/user.go

```diff
@@ -44,3 +45,39 @@ func GenerateRandomBytesSecure(n int) ([]byte, error) {
 
 	return b, nil
 }
+
+// As a safety precaution we don't want to expose the full email address via the API
+// We can rest assured that even if a user session ever leaks, no personal data is provided via api that could link the users addresses or validators to them
+func CensorEmail(mail string) string {
+	parts := strings.Split(mail, "@")
+	if len(parts) != 2 { // invalid mail, should not happen
+		return mail
+	}
+	username := parts[0]
+	domain := parts[1]
+
+	if len(username) > 2 {
+		username = string(username[0]) + "***" + string(username[len(username)-1])
+	}
+
+	// Also censor domain part for not well known domains as they could be used to identify the user if it's a niche domain
+	domainParts := strings.Split(domain, ".")
+	if len(parts) == 2 {
+		// https://email-verify.my-addr.com/list-of-most-popular-email-domains.php
+		wellKnownDomains := []string{"gmail", "hotmail", "yahoo", "apple", "aol", "outlook", "gmx", "live", "comcast", "msn"}
+		isWellKnownDomain := false
+
+		for _, domain := range wellKnownDomains {
+			if domain == domainParts[0] {
+				isWellKnownDomain = true
+				break
+			}
+		}
+
+		if !isWellKnownDomain && len(domainParts[0]) > 2 {
```

Comment:
This can be done with the std lib
```suggestion
		if !slices.Contains(wellKnownDomains, domainParts[0]) && len(domainParts[0]) > 2 {
```